### PR TITLE
Migrate to storybook v7 (#507

### PR DESCRIPTION
**Pull request by joecowton1**
## What are you changing?

Upgrading storybook packages to v7 using their migration tool. 

https://storybook.js.org/docs/react/migration-guide

This has also involved a number of manual changes:
- aligning React versions across projects and adding React where required
- updating storybook-start and storybook-build scripts to use the new `storybook` binary
- adding linting rule exceptions for the palette generator file

## Why?

- we want to be using the latest and greatest

### CODE REVIEW

File: libs/@guardian/source-foundations/.eslintrc.js [hunk 1]
```diff
@@ -45,5 +45,14 @@ module.exports = {
...
 		},
+		{
+			// this is a storybook file, so we don't need to check it so rigorously
+			files: ['storybookColorPalette.tsx'],
+			rules: {
+				'@typescript-eslint/no-unsafe-call': 'off',
+				'@typescript-eslint/no-unsafe-assignment': 'off',
+				'@typescript-eslint/no-unsafe-return': 'off',
+			},
+		},
```
[the remark this answers is not in the document]
Comment (joecowton1) [reply libs/@guardian/source-foundations/.eslintrc.js]: Lo and behold i removed the rules and the errors have vanished, must have been vscode being flakey.

[the remark this answers is not in the document]
Comment (jamesmockett) [reply libs/@guardian/source-foundations/.eslintrc.js]: Not needing them at all is even better 😁

**Review by jamesmockett — APPROVED** [review]
Looks good to me 👍

**Comment by mxdvl** [comment]
This looks great!

Should we be using Preact in Storybook instead of React, as this is what we use in production? https://www.npmjs.com/package/@storybook/preact

**Pull request merged**